c/serverStatusService.c

```diff
@@ -0,0 +1,272 @@
+  memcpy(context->productVersion, productVer, sizeof(context->productVersion) - 1);
+  httpService->userPointer = context;
+  registerHttpService(server, httpService);
+  return 0;
+}
+
+void respondWithServerConfig(HttpResponse *response, JsonObject* config){
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  jsonStartObject(out, "options");
+  jsonPrintObject(out, config);
+  jsonEndObject(out);
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+void respondWithServerRoutes(HttpResponse *response){
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  jsonStartArray(out, "links");
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/config");
+  jsonAddString(out, "rel", "config");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/log");
+  jsonAddString(out, "rel", "log");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/logLevels");
+  jsonAddString(out, "rel", "logLevels");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/environment");
+  jsonAddString(out, "rel", "environment");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonEndArray(out);
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+void respondWithLogLevels(HttpResponse *response, ServerAgentContext *context){
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  JsonObject *logLevels = jsonObjectGetObject(context->serverConfig, "logLevels");
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  if(logLevels){
+    jsonPrintObject(out, logLevels);
+  }
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+void respondWithServerEnvironment(HttpResponse *response, ServerAgentContext *context){
+  /*Information about parameters for smf_unc: https://www.ibm.com/support/knowledgecenter/SSLTBW_2.1.0/com.ibm.zos.v2r1.erbb700/smfp.htm#smfp*/
+  struct utsname unameRet;
+  char hostnameBuffer[256];
+  char* buffer;
+  time_t curr_time;
+  char* c_time_str;
+  int rc = 0;
+  int reqtype = 0x00000005; //fullword. request type
+  int rectype = 0x0000004F; //SMF record type, only type 79 is supported
+  int subtype = 0x00000009; //SMF record subtype
+  int bufferlen = 716800; // Yes the SMF buffer is actually 700Kb roughly
+  int cpuUtil = 0x00000000;
+  int demandPaging = 0x00000000;
+  int options = 0x00000000;
+  int mvsSrm = 0x00000000;
+  int zaapUtil = 0x00000000;
+  int ziipUtil = 0x00000000;
+  uname(&unameRet);
```

Comment:
You could really just implement BPXUNA and make this compatible with LE and METAL. @ifakhrutdinov what do you think?

Reply:
I cannot find a function or service named BPXUNA

Reply:
We need to think how to wrap BPX calls. If @timgerstel does that, we'll need a longer discussion. I think we should do that later. ```uname``` is probably BPX1UNA under the covers.

Reply:
Yeah, it is.